www/pages/[pkg]/[mod]/[symbol].tsx

```diff
@@ -1,186 +1,89 @@
-      symbol: find(mod.functions),
-    },
-    {
-      kind: "variable",
-      symbol: find(mod.variables),
-    }
-  );
-  if (sym === undefined) {
-    return { notFound: true };
-  }
-
-  return {
-    props: {
-      pkg,
-      mod,
-      symbol: sym,
-    },
-  };
-};
-
-export const getStaticPaths: GetStaticPaths = () => {
-  const idx = docs.getPackageIndex();
-
-  const paths: { params: { pkg: string; mod: string; symbol: string } }[] = [];
-  function pushParams(
-    pkg: docs.Pkg,
-    mod: docs.Module,
-    thing: docs.Class | docs.Func | docs.Variable
-  ) {
-    paths.push({
-      params: {
-        pkg: pkg.name,
-        mod: mod.name,
-        symbol: withoutPrefix(mod.name, thing.name),
-      },
-    });
-  }
-
-  idx.forEach((pkg) => {
-    pkg.modules.forEach((mod) => {
-      mod.classes.forEach((cls) => pushParams(pkg, mod, cls));
-      mod.functions.forEach((fn) => pushParams(pkg, mod, fn));
-      mod.variables.forEach((v) => pushParams(pkg, mod, v));
-    });
-  });
-
-  return {
-    paths,
-    fallback: false,
-  };
-};
-
-export default function Page({ pkg, mod, symbol }: Props) {
-  const Sidebar = (() => {
-    switch (symbol.kind) {
-      case "class":
-        return (
-          <>
-            <ClassContents
-              pkg={pkg}
-              mod={mod}
-              cls={symbol.symbol}
-              currentSymbol=""
-            />
-            <ModuleContents pkg={pkg} mod={mod} currentSymbol="" />
-          </>
-        );
-      case "function":
-        return (
-          <ModuleContents
-            pkg={pkg}
-            mod={mod}
-            currentSymbol={symbol.symbol.name}
-          />
-        );
-      case "variable":
-        return (
-          <ModuleContents
-            pkg={pkg}
-            mod={mod}
-            currentSymbol={symbol.symbol.name}
-          />
-        );
-    }
-  })();
+export default function ModulePage() {
+  const router = useRouter();
+  const pkgName = router.query.pkg as string;
+  const modName = router.query.mod as string;
+  const symName = router.query.symbol as string;
+  return (
+    <FetchPackage
+      name={pkgName}
+      content={(pkg) => {
+        const [Sidebar, Content] = (() => {
+          const mod = pkg.modules.find((mod) => mod.name === modName);
+          if (!mod) {
+            return [<ModuleList pkg={pkg} />, `Module ${modName} not found :O`];
+          }
+
+          const cls = mod.classes.find(
+            (cls) => withoutPrefix(mod.name, cls.name) === symName
+          );
+          const func = mod.functions.find(
+            (func) => withoutPrefix(mod.name, func.name) === symName
+          );
+          const variable = mod.variables.find(
+            (variable) => withoutPrefix(mod.name, variable.name) === symName
+          );
+          const symbol = cls || func || variable;
+
+          if (!symbol) {
+            return [<ModuleList pkg={pkg} />, `Symbol ${symName} not found :O`];
```

Comment:
Yep, definitely needs DRYing
```suggestion
            return [<ModuleList pkg={pkg} />, `Symbol ${symName} not found 🤪`];
```

Reply:
My opinion remains:

eh